python/afdko/buildcff2vf.py

```diff
@@ -2,996 +2,498 @@
-from the default master source font. The default font is flagged in the
-designspace file by having the element "<info copy="1" />" in the <source>
-element.
-
-The width values and the GPOS positioning data are drawn from all the
-master source fonts, so each must be built with with a full set of GPOS
-features.
-
-The companion script buildmasterotfs.py will build the master source OTFs
-from the designspace file.
-
-Any python interpreter may be used to run the script, as long as it has
-installed the latest version of the fonttools module from
-https://github.com/fonttools/fonttools
-"""
-
-import collections
-import io
-import logging
-import os
-import sys
-
+import argparse
+from copy import deepcopy
 from fontTools import varLib, version as fontToolsVersion
-from fontTools.misc.py23 import tobytes
-from fontTools.ttLib import TTFont, newTable
-from fontTools.cffLib import (VarStoreData, buildOpcodeDict,
-                              privateDictOperators)
+from fontTools.cffLib.specializer import commandsToProgram
+from fontTools.designspaceLib import DesignSpaceDocument
+from fontTools.misc.fixedTools import otRound
 from fontTools.misc.psCharStrings import T2OutlineExtractor, T2CharString
-from fontTools.pens.t2CharStringPen import T2CharStringPen
-from fontTools.ttLib.tables import otTables
-try:
-    import xml.etree.cElementTree as ET
-except ImportError:
-    import xml.etree.ElementTree as ET
+from fontTools.ttLib import TTFont
+from fontTools.varLib.cff import CFF2CharStringMergePen
+import logging
+import os
 from pkg_resources import parse_version
+import re
+import sys
 
-XML = ET.XML
-XMLElement = ET.Element
-xmlToString = ET.tostring
+__version__ = '1.15.0'
 
-kMaxStack = 48
-kTempCFFSuffix = ".temp.cff"
-kTempCFF2File = "test.cff2"
 
-# setup basic logging to enable fontTools errors to peek thru
-logging.basicConfig()
+# set up for printing progress notes
+def progress(self, message, *args, **kws):
+    # Note: message must contain the format specifiers for any strings in args.
+    level = self.getEffectiveLevel()
+    self._log(level, message, args, **kws)
 
 
-class ACFontError(KeyError):
-    pass
+PROGRESS_LEVEL = logging.INFO+5
+PROGESS_NAME = "progress"
+logging.addLevelName(PROGRESS_LEVEL, PROGESS_NAME)
+logger = logging.getLogger(__name__)
+logging.Logger.progress = progress
 
 
-class AxisMapError(KeyError):
+def getSubset(subset_Path):
+    if not os.path.exists(subset_Path):
+        raise ValueError("could not find subset file path", subset_Path)
+    fp = open(subset_Path, "rt")
+    text_lines = fp.readlines()
+    locationDict = {}
+    cur_key_list = None
+    for li, line in enumerate(text_lines):
+        idx = line.find('#')
+        if idx >= 0:
+            line = line[:idx]
+        line = line.strip()
+        if not line:
+            continue
+        if line[0] == "(":
+            cur_key_list = []
+            location_list = eval(line)
```

Comment:
Regarding the Codacy issue on this part of the code, if `line` is something like this:
```
line = "(42)"
```
then I suggest replacing the use of `eval()` with something like this:
```
location_list = int(line.strip("()"))
```